Expand "Local development" (#363)
architect/arc.codes: src/views/docs/en/guides/developer-experience/local-development.md

> 👉 Note: this example uses Node.js conventions but the process is similar for the [Ruby](/docs/en/reference/runtime/ruby) (bundler) and [Python](/docs/en/reference/runtime/python) (pip) runtimes.

### Update Architect's configuration

To create a new event, add an entry to your manifest's [`@events` pragma](/docs/en/reference/app.arc/events).

Sample `app.arc` file with a new `hit-counter` event:

```arc
# app.arc
@app
my-app

@events
hit-counter
```

> 💡 If the pragma does not already exist, add "@events" on a new line.

### Scaffold the new resource with `arc init`

The Architect CLI [`init` command](/docs/en/reference/cli/init) can be used to create scaffolding for the new event. From the project root:

```console
arc init
```

> 👀 Notice that `arc` finds your updated Architect manifest and creates new project files for the hit-counter event. `arc init` can be run as needed while developing your application.

The following structure was added your project:
<!-- unsure if this diagram is necessary -->
```
.
├── src
│   └── events
│       └── hit-counter
│           ├── index.js
│           └── config.arc
│

> └── app.arc

Reviewer: Hmm, some funky unicode artifacts here?
![image](https://user-images.githubusercontent.com/200964/128097791-4d618aaa-d70f-4144-8034-6a156d46b928.png)

Author: Super weird. that line is the same as the other `└── app.arc` instances in project-layout.md and queues.md.  
and the `─` character that shows up garbled in your screenshot is used in each of the lines above the app.arc line in your screenshot.

The issue must be between the code and the HTML output, maybe the Markdown compiler. but I'm having a hard time reproducing that garbled text.  
I'm already taking a look there since there seems to be an issue with my highlighter.js change, so I can take some time to look more at this.

```

### Add dependencies (optional)

> Your new event may require some dependencies. For this example, [@architect/functions](https://github.com/architect/functions) will be helpful to handle the event subscription. Dependencies can be installed directly to the `hit-counter` directory:

Reviewer 2: Maybe we can link to the [dependency management guide](https://arc.codes/docs/en/guides/developer-experience/dependency-management) in here somehow to note the two approaches to dependency management? 
Perhaps something like:

```suggestion
Your new event may require some dependencies. For this example, [the @architect/functions runtime helper library](../../reference/runtime/node.js) will be helpful to handle the event subscription. [Dependencies in node.js can be managed](dependency-management) either globally from the root `package.json`, or managed via separate `package.json` files in each function's directory. In this example, let's manage them directly in the event's `src/events/hit-counter` directory:
```

> 1. Inside `./src/events/hit-counter` create a `package.json` file with an empty `dependencies` entry:

Reviewer: Good news, we do Lambda treeshaking now! No need to have individual `package.json` files, just add `@architect/functions` to root `package.json` and we'll pick it up at hydration / deploy time.

```json
// package.json
{
  "dependencies": {}
}
```

2. Install dependencies with `npm`:

```console
cd src/events/hit-counter
npm install @architect/functions
```

> 3. From the project root start up the sandbox if it is not already running:

Reviewer 2: ```suggestion
3. From the project root start up the [sandbox](../../reference/cli/sandbox) if it is not already running:
```

```console
arc sandbox
```

> Architect will hydrate your [shared code](/docs/en/guides/developer-experience/sharing-code) and run the updated Architect project.

Reviewer: We didn't mention adding shared code as part of this local dev guide, so this is a bit awkward. But it'd def make sense to demonstrate shared code.

Reviewer 2: Agreed, perhaps we can reword this to state that _sandbox_ will hydrate function code and link to the `arc hydrate` docs while we are at it?

```suggestion
[`arc sandbox`](../../reference/cli/sandbox) will automatically [hydrate - that is, install needed dependencies](../../reference/cli/hydrate) - for every function making up your Architect project.
```

> 📖 Read more about [dependency management](/docs/en/guides/developer-experience/dependency-management).

## Debugging

With an Architect project running locally, there are several ways to work on your application.

You can run a front-end application from the same Architect project to communicate with the back-end or use a client to interface with HTTP functions.

> Of course, the best way to catch bugs is by [testing your Architect project](/docs/en/guides/developer-experience/testing).

Reviewer 2: ```suggestion
Of course, the best way to catch bugs is by [testing your Architect project](testing).
```
